Xext/xselinux_ext.c

```diff
@@ -82,32 +82,27 @@ ProcSELinuxQueryVersion(ClientPtr client)
 static int
 SELinuxSendContextReply(ClientPtr client, security_id_t sid)
 {
-    char *ctx = NULL;
-    int len = 0;
+    x_rpcbuf_t rpcbuf = { .swapped = client->swapped, .err_clear = TRUE };
 
+    int len = 0;
     if (sid) {
+        char *ctx;
         if (avc_sid_to_context_raw(sid, &ctx) < 0)
             return BadValue;
-        len = strlen(ctx) + 1;
+        x_rpcbuf_write_string_0t_pad(&rpcbuf, ctx);
+        len = strlen(ctx);
```

Comment:
Can we get `len` from `rpcbuf` instead of calling `strlen`? Also it was strlen + 1 (counting the terminating zero) - was it incorrect?

Reply:
Not exactly, because rpcbuf the string is written with padding, but the context_len field tells the length of the string (w/o padding).
But i've just seen I had a bug: the terminating zero is also counted.